### PR TITLE
Fix screenreader compatibility issues with the toggle control

### LINKED ISSUE #1619: Toggle doesn't announce when it is changed
Environment: IE11 and JAWS 18
Are you willing to submit a PR to fix? No. I need help
Requested priority: High

Toggle does not say when it is turned "off" or "on". Specifically, it should announce "<label name><state>"

### PR DESCRIPTION
**Pull request by scriby**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #1619 

#### Description of changes

This changes the toggle control to use an invisible checkbox instead of an invisible button. There are no aesthetic changes; it only affects how screen reader users interact with the control.

This approach works better for JAWS and VoiceOver because they announce when the value of checkboxes change, but they do not announce when aria-pressed changes.

Note that the previous approach of using the button and aria-pressed is more semantic and should be preferred at the point there is more universal screen reader support. Alternatively, we could look into some solutions using aria-live to manually announce toggle changes if we wanted to keep with that approach, but it's hard for me to say whether that's really better or worse than using the checkbox.

I tested the changes in JAWS + IE11, JAWS + Chrome, FF + NVDA, Narrator + Edge, and VoiceOver + Chrome on OSX.

Thanks,

Chris

### CODE REVIEW

**Comment by betrue-final-final** [comment]
Thanks @scriby!

**Comment by micahgodbolt** [comment]
@cschlechty care to weigh in on this before we merge these changes?

File: packages/office-ui-fabric-react/src/components/Toggle/Toggle.tsx [hunk 1]
```diff
@@ -122,4 +131,16 @@ export class Toggle extends BaseComponent<IToggleProps, IToggleState> implements
       onChanged(!isChecked);
     }
   }
+
+  @autobind
+  private _onInputKeyDown(ev: React.KeyboardEvent<HTMLElement>) {
+    switch (ev.which) {
+      case KeyCodes.enter:
```
Comment (cschlechty) [inline packages/office-ui-fabric-react/src/components/Toggle/Toggle.tsx]: Should support space as well.

Comment (scriby) [inline packages/office-ui-fabric-react/src/components/Toggle/Toggle.tsx]: Space already works natively, no need to support it directly.

**Comment by scriby** [comment]
The behavior of the enter key is really tricky on this one.

For unsighted users, enter should not work because it's a checkbox.
For sighted (keyboard-only) users, enter should work because it's a toggle.

One angle is that toggle is not an element that should be in a form. The Toggle docs indicate they are for immediate state changes and should not require an additional button press to trigger.

The other angle is that if we look at the previous solution (with button + aria-pressed), what would have happened if the toggle was placed in a form and enter was pressed? It seems that emulating that behavior would be ideal, unless it was "broken" to begin with. I'd consider "broken" to mean that both the toggle changed **and** the form submitted.

**Pull request merged**